pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -875,7 +879,7 @@ public static CompletableFuture<ClusterDataImpl> checkLocalOrGetPeerReplicationC
                 if (!allowDeletedNamespace && policies.deleted) {
                     String msg = String.format("Namespace %s is deleted", namespace.toString());
                     log.warn(msg);
-                    validationFuture.completeExceptionally(new RestException(Status.PRECONDITION_FAILED,
+                    validationFuture.completeExceptionally(new RestException(Status.NOT_FOUND,
```

Comment:
It's incorrect here.
If `allowDeletedNamespace` is false. We should return `Status.PRECONDITION_FAILED`. No matter if `policies.deleted` is true or false.

Reply:
If we return `Status.PRECONDITION_FAILED` here, the error code will same as the scenario `if (policies.replication_clusters.isEmpty()`, then we can not determine whether the error was caused by `namespace deleted` when calling this method

Reply:
Oh, got it. I misunderstand the `allowDeletedNamespace`. I thought it was `forceDeleteNamespaceAllowed=false` in broker.conf.